src/utils/cache.test.ts

```diff
@@ -0,0 +1,157 @@
+import { blockTagParamIndex, cacheTypeForMethod, canCache } from './cache';
+
+const knownMethods = [
+  'web3_clientVersion',
+  'web3_sha3',
+  'eth_protocolVersion',
+  'eth_getBlockTransactionCountByHash',
+  'eth_getUncleCountByBlockHash',
+  'eth_getCode',
+  'eth_getBlockByHash',
+  'eth_getTransactionByHash',
+  'eth_getTransactionByBlockHashAndIndex',
+  'eth_getTransactionReceipt',
+  'eth_getUncleByBlockHashAndIndex',
+  'eth_getCompilers',
+  'eth_compileLLL',
+  'eth_compileSolidity',
+  'eth_compileSerpent',
+  'shh_version',
```

Comment:
`shh_version` doesn't exist anymore.

tried calling infura with it:

<img width="1665" alt="image" src="https://user-images.githubusercontent.com/364566/215529428-bf418e3b-bec5-46e3-92e6-f8786ed83206.png">

There are more too, `eth_compileLLL` and the other `compileX` dont exist anymore either.

we should probably get our method lists for eth rpc from the spec: https://github.com/ethereum/execution-apis

Reply:
By "known methods" here, I meant known by the middleware rather than us. This list is taken directly from the code under test.

Definitely lots of updates needed to this middleware but I'd rather not take that work on right now. Not until after we finish this multichain work.